src/main/java/server/RequestParser.java

```diff
@@ -0,0 +1,72 @@
+package server;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.util.HashMap;
+
+public class RequestParser {
+    private BufferedReader in;
+
+    public RequestParser(BufferedReader in) {
+        this.in = in;
+    }
+
+    public HashMap parse() {
+        HashMap parsedRequest = new HashMap();
+
+        parsedRequest.put("requestLine", parseRequestLine());
+
+        HashMap<String, String> requestHeaders = parseRequestHeaders();
+        parsedRequest.put("requestHeaders", requestHeaders);
+
+        Integer contentLength = parseRequestBodyContentLength(requestHeaders);
+        parsedRequest.put("requestBody", parseRequestBody(contentLength));
```

Comment:
I think it's probably better to put `Integer contentLength = parseRequestBodyContentLength(requestHeaders);` inside of `parseRequestBody` and pass in the `requestHeaders` to that function instead.

If we were to ask about the `parse` function's responsibilities, it should probably be to just create the `parsedRequest` hashmap from calling a few private functions for each key. However, line 22 probably makes this divert from that and adds an extra responsibility that can be a part of one of the private functions.